src/operator/src/statement/copy_table_from.rs

```diff
@@ -498,6 +515,14 @@ fn can_cast_types_for_greptime(from: &ArrowDataType, to: &ArrowDataType) -> bool
 }
 
 fn ensure_schema_compatible(from: &SchemaRef, to: &SchemaRef) -> Result<()> {
+    if from.fields().len() != to.fields().len() {
+        return error::InvalidHeaderSnafu {
+            table_schema: to.to_string(),
+            file_schema: from.to_string(),
+        }
+        .fail();
+    }
```

Comment:
We can’t simply determine schema incompatibility based on field length. If the source schema is a subset of the target schema, they should be considered compatible.

Reply:
Yes, and I suggest adding more tests to cover this feature, as there are many cases we need to handle.